activation/atxdb.go

```diff
@@ -187,6 +191,14 @@ func (db *ActivationDb) CalcActiveSetFromView(view []types.BlockID, pubEpoch typ
 // - ATX LayerID is NipstLayerTime or less after the PositioningATX LayerID.
 // - The ATX view of the previous epoch contains ActiveSetSize activations.
 func (db *ActivationDb) SyntacticallyValidateAtx(atx *types.ActivationTx) error {
+	pub, err := types.ExtractPublicKey(atx)
+	if err != nil {
+		return fmt.Errorf("cannot validate atx sig atx id %v err %v", atx.ShortString(), err)
+	}
+	err = db.ValidateSignedAtx(*pub, atx)
```

Comment:
Why aren't you skipping this validation if there is an empty prev atx? I know that `ValidateSignedAtx` will return nil, but it makes this code not straight forward for the reader of the code which might assume that you're validating the sig also in the first atx